pwndbg/search.py

```diff
@@ -58,7 +72,16 @@ def search(searchfor, mappings=None, start=None, end=None, executable=False, wri
             if length <= 0:
                 break
 
-            start = i.search_memory(start, length, searchfor)
+            try:
+                start = i.search_memory(start, length, searchfor)
+            except gdb.error:
+                # While remote debugging on an embedded device and searching
+                # through a large memory region (~512mb), gdb may return an error similar
+                # to `error: Invalid hex digit 116`, even though the search
+                # itself is ok. It seems to have to do with a timeout.
+                print("WARN: gdb.search_memory failed, stopping early...")
```

Comment:
Ideally this should be a separate PR/commit if possible, but I'm okay leaving this here.

On the other hand, should we maybe check the gdb.error message string or print it to the user as well?

If we think this may be a timeout, we could say it to the user as well that this may happen for embedded debugging

Reply:
I'll break it out into a separate PR then. 

Confusingly the gdb error is `error: Invalid hex digit 116`, but from a bit of digging it seems like this sometimes comes from gdb getting confused during timeouts. The unfortunate thing is that error isn't exactly useful to relay back to the user, but I guess we may as well reported back.